botocore/signers.py

```diff
@@ -492,7 +492,7 @@ def generate_presigned_url(self, ClientMethod, Params=None, ExpiresIn=3600,
 
     # Prepare the request dict by including the client's endpoint url.
     prepare_request_dict(
-        request_dict, endpoint_url=self.meta.endpoint_url)
+        request_dict, endpoint_url=self.meta.endpoint_url, context={})
```

Comment:
This is kind of a weird idiom to pass an empty dict through to the prepare function if we don't need to prepare/reference it before hand.  We should just let the `prepare_request_dict` own creation of the context dict.